src/mpid/ch4/netmod/ucx/ucx_win.h

```diff
@@ -133,7 +159,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_rma_win_cmpl_hook(MPIR_Win * win)
     if (MPIDI_UCX_is_reachable_win(win) && MPIDI_UCX_win_need_flush(win)) {
         ucs_status_t ucp_status;
         /* maybe we just flush all eps here? More efficient for smaller communicators... */
-        ucp_status = ucp_worker_flush(MPIDI_UCX_global.ctx[0].worker);
+        ucp_status = MPIDI_UCX_flush(0);
```

Comment:
Is this change necessary to enable multi-vni pt2pt and collectives?

Reply:
See this comments: https://github.com/pmodels/mpich/pull/4648#issuecomment-651451756

Somehow `ucp_worker_flush` hangs in the test `./rma/rget_unlock 2` unless global progress is polled even though only vci 0 is ever used. I don't quite understand why yet. Replacing `ucp_worker_flush` with the non-blocking version allows polling global progress in it to prevent the deadlock.

Reply:
The old blocking flush function calls this: https://github.com/openucx/ucx/blob/master/src/ucp/rma/rma.inl#L53-L74, which is very similar to `MPIDI_UCX_flush` but only progresses a single worker. I'm concerned that we don't understand what is going on here. I don't think this is ready to be merged. We will talk about this on the call Thursday, but the release is going to be delayed significantly, so this still has time to investigate and get it right.

Reply:
Sounds good.

Reply:
I found the root issue. When ucx creating workers and establishing endpoints (in `all_vnis_address_exchange` ), it is in fact non-blocking operations and it only can be completed/cleared by progress on all workers that are involved. Since we are doing all-to-all wireup, that means we need do progress on all vni workers. UCX internally only maintain a single `flush_ops_count` for each worker. So even though in the test program we only use a single vni and the rma operations are actually cleared, the pending ops on connection to other vni prevents the `ucp_worker_flush` to clear.

The fix is to force flush all workers during `MPIDI_UCX_post_init`.

Reply:
@rzambre Refer to my comment above. In your case since you are only doing one-to-one connections rather than all-to-all connections, you won't have the same issue.